packages/react/src/components/Popover/index.tsx

```diff
@@ -14,13 +20,47 @@ export const Popover = ({
   onClickOutside,
   borderless = false,
 }: PopoverProps) => {
+  const popoverRef = useRef<HTMLDivElement>(null);
   useHandleClickOutside(parentRef, onClickOutside);
+  const [topPosition, setTopPosition] = useState(0);
+
+  useLayoutEffect(() => {
+    const handleWindowResize = () => {
+      if (!parentRef.current || !popoverRef.current) return;
+
+      const { offsetTop: parentOffsetTop, clientHeight: parentClientHeight } =
+        parentRef.current;
+      const { clientHeight: popoverClientHeight } = popoverRef.current;
+
+      if (
+        parentOffsetTop + parentClientHeight + popoverClientHeight >
+          window.innerHeight &&
+        parentOffsetTop - popoverClientHeight > 0
+      ) {
+        setTopPosition(-popoverClientHeight);
+      } else {
+        setTopPosition(parentClientHeight);
+      }
+    };
+
+    window.addEventListener("resize", handleWindowResize);
+
+    handleWindowResize();
+    return () => {
+      window.removeEventListener("resize", handleWindowResize);
+    };
+  }, []);
```

Comment:
I'm afraid this will not work properly until the parent container has a `position` set. Otherwise the dialog will be positioned on top of the document.

The popover story is good example of what is wrong.
<img width="234" alt="image" src="https://github.com/openfun/cunningham/assets/9265241/e6ed0b0a-1555-48aa-8963-9f8a4ef63677">

Reply:
Well seen:

Reply:
https://github.com/openfun/cunningham/pull/133/commits/35240d76f114fb094d46c837d7c6a3c78738f2aa

Reply:
It is needed for the integration of the datepicker in Marsha but we could add this feature optional, let me know.